### PR TITLE
Brawl match bugfixes

### LINKED ISSUE #5: Close all open views on //match stop

### LINKED ISSUE #7: Ensure all maps are on server before starting match

### LINKED ISSUE #17: When having multiple matches after each other, the map counter is not reset

### PR DESCRIPTION
**Pull request by JosVroegindeweij**
Resolves #5 
Resolves #7 
Resolves #17

### CODE REVIEW

File: pyplanet/apps/contrib/brawl_match/__init__.py [hunk 1]
```diff
@@ -39,6 +39,7 @@ def __init__(self, *args, **kwargs):
 		self.backup_script_name = None
 		self.backup_settings = None
 		self.maps_played = 0
+		self.views_open = []
```
Comment (TheSuperNoob) [inline pyplanet/apps/contrib/brawl_match/__init__.py]: Maybe open_views is a better name? Not sure what is proper english but open_views sounds better when i say it out loud.

File: pyplanet/apps/contrib/brawl_match/views.py [hunk 2]
```diff
@@ -65,6 +65,10 @@ def __init__(self, app, maps):
 		return items
 
 
+	def __del__(self):
+		self.app.views_open.pop(self)
```
Comment (TheSuperNoob) [inline pyplanet/apps/contrib/brawl_match/views.py]: Does pop take an object as an argument? I thought it only worked with indices.

File: pyplanet/apps/contrib/brawl_match/__init__.py [hunk 3]
```diff
@@ -74,9 +74,14 @@ def __init__(self, *args, **kwargs):
 	async def start_match_command(self, player, *args, **kwargs):
 		if self.match_tasks:
 			await self.brawl_chat(f'A match is currently in progress!', player)
+		elif not await self.check_maps_on_server():
+			await self.brawl_chat(f'Not all maps for the match are on the server', player)
 		else:
 			await self.register_match_task(self.start_match, player)
 
+	async def check_maps_on_server(self):
+		return all(map(lambda t: self.instance.map_manager.playlist_has_map(t[0]), self.brawl_maps))
```
Comment (TheSuperNoob) [inline pyplanet/apps/contrib/brawl_match/__init__.py]: `return all([self.map_manager.playlist_has_map(map) for map in self.brawl_maps])`

Generally list comprehensions are better Python [solutions](https://stackoverflow.com/questions/1247486/list-comprehension-vs-map) compared to `map` and `filter` functions.

**Review by TheSuperNoob — CHANGES_REQUESTED** [review]

File: pyplanet/apps/contrib/brawl_match/views.py (hunk 2 above)
Comment (TheSuperNoob) [reply pyplanet/apps/contrib/brawl_match/views.py]: Ah, i see you changed it to .remove :)

File: pyplanet/apps/contrib/brawl_match/__init__.py [hunk 4]
```diff
@@ -199,14 +207,19 @@ def __init__(self, *args, **kwargs):
...
 			return
+
+		await self.brawl_chat(f'Admin {player.nickname}$z$fff stopped match!')
 		for task in self.match_tasks:
 			if not task.done():
 				task.cancel()
 		self.match_tasks = []
-		await self.brawl_chat(f'Admin {player.nickname}$z$fff stopped match!')
+
+		map(lambda v: await v.destroy, self.views_open)
```
Comment (TheSuperNoob) [inline pyplanet/apps/contrib/brawl_match/__init__.py]: `(await view.destroy()) for view in self.view_open`
Should work too I'm pretty sure. But test it first to make sure.

Comment (JosVroegindeweij) [reply pyplanet/apps/contrib/brawl_match/__init__.py]: That syntax is not allowed it seems :D

Comment (TheSuperNoob) [reply pyplanet/apps/contrib/brawl_match/__init__.py]: Then i would use a full for loop. IE `for view in self.views_open: view.destroy()`

I know im quite anti `map` here but i really dont like mapping to a lambda function.

File: pyplanet/apps/contrib/brawl_match/__init__.py [hunk 5]
```diff
@@ -80,7 +80,7 @@ def __init__(self, *args, **kwargs):
 			await self.register_match_task(self.start_match, player)
 
 	async def check_maps_on_server(self):
-		return all(map(lambda t: self.instance.map_manager.playlist_has_map(t[0]), self.brawl_maps))
+		return all([self.instance.map_manager.playlist_has_map(map) for (map, timeout) in self.brawl_maps])
```
Comment (TheSuperNoob) [inline pyplanet/apps/contrib/brawl_match/__init__.py:83]: Not really sure whats best here. 
`return all([self.instance.map_manager.playlist_has_map(map[0]) for map in self.brawl_maps])`
`return all([self.instance.map_manager.playlist_has_map(map) for (map, _) in self.brawl_maps])`
or to just leave it as is. They all have their advantages I guess. I myself prefer my first one over the other two, but you can decide for yourself if you want to keep it or change to one of the two i suggested.

**Review by TheSuperNoob — APPROVED** [review]

**Pull request merged**